### PR TITLE
feat: add grouped settings flag

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/1466

#### Changes proposed in this Pull Request

Adding a feature flag to group the changes needed to update the settings.
The feature flag will be used to enable the new designs, as well as enabling the new payment methods.

#### Testing instructions

Assuming you are using the dev version of the plugin:
```
cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools
git pull
git checkout frosso/add-grouped-settings-flag
```

Go to the WC Pay Dev Utils page, enable/disable the flag: https://d.pr/i/sfW0OX

The related PR will also need to be reviewed and merged: 8-gh-Automattic/woocommerce-payments-dev-tools

*

-------------------

- [-] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-payments-admin.php [hunk 1]
```diff
@@ -438,4 +442,13 @@ private function is_tos_agreement_required() {
...
 		return get_option( '_wcpay_feature_account_overview' );
 	}
+
+	/**
+	 * Checks whether the grouped settings feature is enabled
+	 *
+	 * @return bool
+	 */
+	private static function is_grouped_settings_enabled() {
+		return get_option( 'wcpaydev_grouped_settings', '0' ) === '1';
```
Comment (superdav42) [inline includes/admin/class-wc-payments-admin.php:452]: 'yes'/'no' is the [convention](https://github.com/woocommerce/woocommerce/blob/23710744c01ded649d6a94a4eaea8745e543159f/tests/legacy/unit-tests/order/coupons.php#L62) usually used by option to designate boolean values that are actually stored as strings.

Comment (frosso) [reply includes/admin/class-wc-payments-admin.php:452]: The `update_option_from_checkbox` method in the dev tools helper stores these values as `'0'` or `'1'`, so I need to make the same check here

**Review by superdav42 — APPROVED** [review]
LGTM

File: includes/admin/class-wc-payments-admin.php (hunk 1 above)
Comment (waclawjacek) [inline includes/admin/class-wc-payments-admin.php:452]: This is a minor nitpick but it looks like the [`_wcpay_feature_X` format](https://github.com/Automattic/woocommerce-payments/pull/1467/files#diff-3863b4f88d0515e080fadc56af3c5e38e942afb96e569cb000ae6a594e8bea4aR443) is used for feature flags, and `wcpaydev_X` for dev tools-specific settings. The latter seem to be less of feature flags and more of dev adjustments.

Maybe the option should be called `_wcpay_feature_grouped_settings` instead?

Comment (frosso) [reply includes/admin/class-wc-payments-admin.php]: That's a great point, I like it! Done in 759dfc0

**Pull request merged**